Meadowlark-js/packages/meadowlark-core/src/validation/DocumentIdentityExtractor.ts

```diff
@@ -32,8 +32,18 @@ type NullableTopLevelEntity = { superclass: TopLevelEntity | null };
  */
 function singleIdentityFrom(property: EntityProperty, body: object, documentPath: string[]): DocumentIdentity {
   const { apiMapping } = property.data.meadowlark as EntityPropertyMeadowlarkData;
-  const documentPathAsString: string = [...documentPath, apiMapping.fullName].join('.');
-  const elementValue: string | undefined = R.path([...documentPath, apiMapping.fullName], body);
+
+  let path: string[];
+
+  if (property.type === 'schoolYearEnumeration' && property.roleName) {
```

Comment:
Not wrong to do a roleName check this way but the type of EntityProperty.roleName (like almost everything in the MetaEd model) is not nullable. Since the string is not nullable, the absence of a string value in the MetaEd model is the empty string, which simplifies these kind of checks.